sympy/vector/functions.py

```diff
@@ -474,3 +474,54 @@ def _path(from_object, to_object):
         from_path.append(other_path[i])
         i -= 1
     return index, from_path
+
+def orthogonalize(vlist, orthonormal=False):
+    """
+    Takes a sequence of independent vectors and orthogonalizes them
+    using the Gram - Schmidt process. Returns a list of
+    orthogonal or orthonormal vectors.
+
+    Parameters
+    ==========
+
+    vlist : list of independent vectors to be made orthogonal.
+
+    orthonormal : Optional parameter
+    Set to True if the the vectors returned should be orthonormal.
+    Default: False
+
+    Examples
+    ========
+
+    >>> from sympy.vector.coordsysrect import CoordSysCartesian
+    >>> from sympy.vector.vector import Vector, BaseVector
+    >>> from sympy.vector.functions import orthogonalize
+    >>> C = CoordSysCartesian('C')
+    >>> i, j, k = C.base_vectors()
+    >>> v1 = i + 2*j
+    >>> v2 = 2*i + 3*j
+    >>> orthogonalize(v1, v2)
+    [C.i + 2*C.j, 2/5*C.i + (-1/5)*C.j]
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Gram-Schmidt_process
+
+    """
+
+    if not all(isinstance(vec, Vector) for vec in vlist):
+        raise TypeError('Each element must be of Type Vector')
+
+    ortho_vlist = []
+    for i, term in enumerate(vlist):
+        for j in range(i):
+            term -= ortho_vlist[j].projection(vlist[i])
+        if simplify(term).equals(Vector.zero):
```

Comment:
Please add a comment here saying something like:

`# TODO : The following line introduces a performance issue and needs to be changed once a good solution for issue #XXXX is found.`